js/classes.js

```diff
@@ -180,10 +180,14 @@ class Item {
 }
 
 class Module {
+    #levelTextElement
+
     constructor(baseData) {
         this.data = baseData;
         this.title = baseData.title;
         this.components = baseData.components;
+        this.level = 0;

```

Comment:
Das ist doch eigentlich nur virtuell? Halt so 
```
getLevel() {
  return this.components.reduce(function(component, sum) { return sum + component.getLevel() }, 0);
}
```
Und die components können dasselbe für ihre Operations machen.
Doppelte Buchführung würde ich nur machen, wenn klar ist, dass on-the-fly-Berechnungen ein CPU Bottleneck werden.